README.md

```diff
@@ -100,10 +100,9 @@ Our manual shows full [read & write examples](https://openpmd-api.readthedocs.io
 
 Required:
 * CMake 3.15.0+
-* C++14 capable compiler, e.g. g++ 5.0+, clang 5.0+, VS 2017+
+* C++17 capable compiler, e.g., g++ 7+, clang 6+, MSVC 19.15+, icpc 19+, icpx
```

Comment:
I wonder if we should advertise/support only clang 7+, primarily due to the incompatibilities of `<variant>` in earlier versions with GCC's libstdc++ (and to a minor degree: the small false warning in `Filesystem.cpp`)
```suggestion
* C++17 capable compiler, e.g., g++ 7+, clang 7+, MSVC 19.15+, icpc 19+, icpx
```